AzureWebFarm.OctopusDeploy/Infrastructure/OctopusDeploy.cs

```diff
@@ -53,28 +54,52 @@ public void DeleteMachine()
         }
 
         public void DeployAllCurrentReleasesToThisMachine()
-        {
+        {            
+            const string targetRolePropertyName = "Octopus.Action.TargetRoles";
             List<TaskState> currentStatuses;
             try
             {
                 var machineId = _repository.Machines.FindByName(_machineName).Id;
                 var environment = _repository.Environments.FindByName(_config.TentacleEnvironment).Id;
+                var projects = _repository.Projects.FindAll()

```

Comment:
That is what I would assume too.

Do you think it appreciably added to the spin up time?

Did you find these changes fixed your problems when you deployed?

If so happy to merge this pr and get it deployed! :)

> On 24 Apr 2015, at 6:33 pm, Lee Baker <notifications@github.com> wrote:
> 
> In AzureWebFarm.OctopusDeploy/Infrastructure/OctopusDeploy.cs:
> 
> >              List<TaskState> currentStatuses;
> >              try
> >              {
> >                  var machineId = _repository.Machines.FindByName(_machineName).Id;
> >                  var environment = _repository.Environments.FindByName(_config.TentacleEnvironment).Id;
> > +                var projects = _repository.Projects.FindAll()
> My assumption was that this would make one call to get the projects, then one call per project to get the deployment process for that project. I have not watched the traffic to test this however. I had problems running locally so have only tested "in production".
> I wasn't too concerned about the number of API calls because this is only for tentacle start-up.
> 
> —
> Reply to this email directly or view it on GitHub.
>